tests/test_examples.py

```diff
@@ -42,67 +45,86 @@
 
-    # check if z3 dependency is available
-    z3_dependency_available = True
-    try:
-        import z3
-    except ImportError:
-        z3_dependency_available = False
+    if z3_enabled:
+        try:
+            import z3
+        except ImportError:
+            z3_enabled = False
 
     for path in paths:
         for root, _, files in os.walk(path, topdown=True):
             for filename in files:
-                if (
-                    filename not in IGNORED_TESTS
-                    and filename.endswith(".py")
-                    and (z3_dependency_available or filename not in Z3_RELATED_TESTS)
-                ):
-                    full_path = os.path.join(root, filename)
-                    rel_path = os.path.relpath(full_path, path)
-                    test_files.append(os.path.join(path, rel_path))
+                if filename in IGNORED_TESTS or not filename.endswith(".py"):
+                    continue
+                # Filters based on z3_enabled flag
+                if z3_enabled and filename not in Z3_RELATED_TESTS:
+                    continue
+                if not z3_enabled and filename in Z3_RELATED_TESTS:
+                    continue
+
+                full_path = os.path.join(root, filename)
+                rel_path = os.path.relpath(full_path, path)
+                test_files.append(os.path.join(path, rel_path))
 
     return test_files
 
 
-def _symbols_by_file_pyta(paths: list[str], include_msg: bool = False) -> dict[str, set[str]]:
+def _symbols_by_file_pyta(
+    paths: list[str],
+    include_msg: bool = False,
+    z3_enabled: bool = False,
+    pycodestyle_enabled: bool = False,
+) -> dict[str, set[str]]:
     """
     Run python_ta.check_all() on files from specified directories and return the map of file name to the
     set of PythonTA messages it raises. If include_msg is set True, PythonTA message descriptions are
     included along with message symbols.
 
     :param paths: The paths to retrieve the files from.
     :param include_msg: whether to include message descriptions in the symbol set
+    :param z3_enabled: Option to enable z3 dependency.
+    :param pycodestyle_enabled: Option to enable pycodestyle-specific checks (E9989).
     :return: A dictionary mapping each file name to a set of PythonTA message symbols
     (and descriptions if include_msg is True).
     """
     sys.stdout = StringIO()
-    python_ta.check_all(
-        module_name=get_file_paths(paths),
-        config={
-            "output-format": "pyta-json",
-            "enable": ["C9960", "C0103"],
-            "z3": True,
-        },
-    )
-
-    jsons_output = sys.stdout.getvalue()
-    sys.stdout = sys.__stdout__
-    pyta_list_output = json.loads(jsons_output)
+    files_to_check = get_file_paths(paths, z3_enabled=z3_enabled)
+    pyta_list_output = []
+    for file in files_to_check:
+        if pycodestyle_enabled:
+            enable_checks = _PYCODESTYLE_CODES + ["C9960"]
+        elif z3_enabled:
```

Comment:
Similar to my previous Z3-related comment, this branch isn't necessary, nor is defining a separate `_Z3_CODES`. When the Z3 tests are running, we should still only be checking a single check (whatever is specified by the filename, determined by Line 115). `z3_enabled` can be used to set the `z3` config value, as you're currently doing, but should _not_ be determining which checks are run.